tests/v2/validation/connection/workload.go

```diff
@@ -0,0 +1,23 @@
+package connection
+
+import (
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/workloads"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	DeploymentSteveType = "apps.deployment"
+	defaultNamespace    = "default"
+)
+
+func createDeployment(steveclient *steveV1.Client, containersTemplate []corev1.Container, containerName string) (*steveV1.SteveAPIObject, error) {
+	podTemplate := workloads.NewPodTemplate(containersTemplate, []corev1.Volume{}, []corev1.LocalObjectReference{}, nil)
+	deployment := workloads.NewDeploymentTemplate(containerName, defaultNamespace, podTemplate, true, nil)
+	deploymentResp, err := steveclient.SteveType(workloads.DeploymentSteveType).Create(deployment)
+	if err != nil {
+		return nil, err
+	}
+
+	return deploymentResp, err
+}
```

Comment:
Wouldn't this make more sense as an extension that could be reused across the framework?

Reply:
Hi @susesgartner we already have some methods on C:\Users\LeonardoScalabrini\forks\shepherd\extensions\workloads\template.go that we can reuse to create new deployments, we could make this method as extension if in the future this method begin to be used by others tests, what do you think?

Reply:
I would prefer not to wait as I've already started writing a test that will consume this function.

Reply:
For some more context, we use the `snapshots` package could reuse this as well since it utilizes deployments in tests. So that would be another usecase that as an extension would be beneficial.

Reply:
Hi @susesgartner @markusewalker I've rebased the code and now I am reusing the https://github.com/rancher/rancher/blob/main/tests/v2/actions/workloads/deployment/deployment.go CreateDeployment function, so I removed the workload.go file